src/extra/libs/fsdrv/lv_fs_littlefs.c

```diff
@@ -0,0 +1,342 @@
+/**
+ * @file lv_fs_littlefs.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "../../../lvgl.h"
+
+#if LV_USE_FS_LITTLEFS
+#include "lfs.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+#if LV_FS_LITTLEFS_LETTER == '\0'
+    #error "LV_FS_LITTLEFS_LETTER must be an upper case ASCII letter"
+#endif
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+static lfs_t * fs_init(void);
+
+static void * fs_open(lv_fs_drv_t * drv, const char * path, lv_fs_mode_t mode);
+static lv_fs_res_t fs_close(lv_fs_drv_t * drv, void * file_p);
+
+static lv_fs_res_t fs_read(lv_fs_drv_t * drv, void * file_p, void * buf, uint32_t btr, uint32_t * br);
+static lv_fs_res_t fs_write(lv_fs_drv_t * drv, void * file_p, const void * buf, uint32_t btw, uint32_t * bw);
+
+static lv_fs_res_t fs_seek(lv_fs_drv_t * drv, void * file_p, uint32_t pos, lv_fs_whence_t whence);
+static lv_fs_res_t fs_tell(lv_fs_drv_t * drv, void * file_p, uint32_t * pos_p);
+/*static lv_fs_res_t fs_size(lv_fs_drv_t * drv, void * file_p, uint32_t * size_p);*/
+
+static void * fs_dir_open(lv_fs_drv_t * drv, const char * path);
+static lv_fs_res_t fs_dir_read(lv_fs_drv_t * drv, void * dir_p, char * fn);
+static lv_fs_res_t fs_dir_close(lv_fs_drv_t * drv, void * dir_p);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ * GLOBAL PROTOTYPES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+void lv_fs_littlefs_init(void)
+{
+    /*----------------------------------------------------
+     * Initialize your storage device and File System
+     * -------------------------------------------------*/
+    lfs_t * lfs_p = fs_init();
+
+    /*---------------------------------------------------
+     * Register the file system interface in LVGL
+     *--------------------------------------------------*/
+
+    /*Add a simple drive to open images*/
+    static lv_fs_drv_t fs_drv; /*A driver descriptor*/
+    lv_fs_drv_init(&fs_drv);
+
+    /*Set up fields...*/
+    fs_drv.letter = LV_FS_LITTLEFS_LETTER;
+    fs_drv.cache_size = LV_FS_LITTLEFS_CACHE_SIZE;
+
+    fs_drv.open_cb = fs_open;
+    fs_drv.close_cb = fs_close;
+    fs_drv.read_cb = fs_read;
+    fs_drv.write_cb = fs_write;
+    fs_drv.seek_cb = fs_seek;
+    fs_drv.tell_cb = fs_tell;
+
+    fs_drv.dir_open_cb = fs_dir_open;
+    fs_drv.dir_close_cb = fs_dir_close;
+    fs_drv.dir_read_cb = fs_dir_read;
+
+    /*#if LV_USE_USER_DATA*/
+    fs_drv.user_data = lfs_p;
+    /*#endif*/
+
+    lv_fs_drv_register(&fs_drv);
+}
+
+/**********************
+ *   STATIC FUNCTIONS
+ **********************/
+
+/*Initialize your Storage device and File system.*/
+static lfs_t * fs_init(void)
```

Comment:
How to replace it by an external function? 

I think we should have something like
```c
lv_fs_littlefs_set_driver(LV_FS_LITTLEFS_LETTER, &my_lfs);
```

Reply:
At the moment it's done in the main testcode by setting 'user_data' element of lv_fs_drv_t struct to the lfs_t object. 
    lv_fs_drv_t *lfsdrv_p = lv_fs_get_drv('L')
    lfsdrv_p->user_data = lfs_p;
I'll create the above mentioned convenience function for this.

Reply:
Great, thank you! Please also mention the usage in the docs.

Reply:
The convenience function is made. Sure, I'll mention it in the docs, for the moment the description is given in the lv_fs_littlefs.c source-code above the function.